### PR TITLE
Fix calling Dispatcher.emit_startup (#1242)

### PR DESCRIPTION
**Pull request by arynyklas**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

#1242 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: aiogram/dispatcher/dispatcher.py [hunk 1]
```diff
@@ -546,7 +547,7 @@ async def start_polling(
             finally:
                 loggers.dispatcher.info("Polling stopped")
                 try:
-                    await self.emit_shutdown(bot=bots[-1], **workflow_data)
+                    await self.emit_shutdown(bots[-1], **workflow_data)
```
Comment (JrooTJunior) [inline aiogram/dispatcher/dispatcher.py:550]: The same as in startup callback

**Review by JrooTJunior — CHANGES_REQUESTED** [review]

**Comment by JrooTJunior** [comment]
Also please add the short changes description to the `CHANGES/1242.bugfix.rst` file

File: aiogram/dispatcher/event/handler.py [hunk 2]
```diff
@@ -37,6 +38,8 @@ def _prepare_kwargs(self, kwargs: Dict[str, Any]) -> Dict[str, Any]:
         }
 
     async def call(self, *args: Any, **kwargs: Any) -> Any:
+        if "FSMContextMiddleware.close" in str(self.callback):
```
Comment (JrooTJunior) [inline aiogram/dispatcher/event/handler.py:41]: wtf is going on?

**Review by JrooTJunior — APPROVED** [review]

File: aiogram/dispatcher/event/handler.py (hunk 2 above)
Comment (arynyklas) [reply aiogram/dispatcher/event/handler.py:41]: I didn't better solution for this bug.

Comment (JrooTJunior) [reply aiogram/dispatcher/event/handler.py:41]: It is very bad solution and doesn't make anything better.
You should not change this place, your issue can be fixed in two lines at place where which you do `.pop`.

Just check that key is present in workflow_data and only then pop it, but all other changes should be rolled back.

**Comment by JrooTJunior** [comment]
I don't see any activity in this PR, so, i'll remake it.

**Comment by JrooTJunior** [comment]
Fixed in #1247

**Pull request closed without merging**